### PR TITLE
Fix user isn't a participant of report displays in detail

### PR DESCRIPTION
**Pull request by dukenv0307**
### Details

The participant details link work even participant is not in the report
### Fixed Issues

$ https://github.com/Expensify/App/issues/19994
PROPOSAL: https://github.com/Expensify/App/issues/19994#issuecomment-1572608393

### Tests

1. Open the app and login with user A
2. Open the app on another device and login with user B
3. From user A device, open #announce of workspace in which user B is admin
4. Click on #announce in header
5. Click on members
6. Click on any member to open its profile page
7. Verify that the URL is `/a/:accountID` instead of `/r/:reportID/participants/a/:accountID`
8. From user B, Go to  Settings --> Workspaces, select the workspace used in Step 3
9. Click on Members,  select member used in step 6, remove the member
10. On user A device, reload the profile page 
11. Verify that the user information is hidden.
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as above
### QA Steps

1. Open the app and login with user A
2. Open the app on another device and login with user B
3. From user A device, open #announce of workspace in which user B is admin
4. Click on #announce in header
5. Click on members
6. Click on any member to open its profile page
7. Verify that the URL is `/a/:accountID` instead of `/r/:reportID/participants/a/:accountID`
8. From user B, Go to  Settings --> Workspaces, select the workspace used in Step 3
9. Click on Members,  select member used in step 6, remove the member
10. On user A device, reload the profile page 
11. Verify that the user information is hidden.
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

[Screencast from 29-06-2023 18:40:50.webm](https://github.com/Expensify/App/assets/129500732/7f21d024-6aca-43b9-8afa-9e5684c8954b)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/129500732/79b9d95f-bfc6-4f04-b7ff-de2c46c05808

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/129500732/927e3b5f-1c81-4407-af94-f1acc31dd1b1

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/129500732/9a3fa32a-3b26-4c45-a65c-0f095a4a08ac

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/129500732/db339892-dbd2-419e-9244-11765641897c

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/129500732/295b38af-4536-440d-9857-7a50a7987e57

</details>

### CODE REVIEW

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by mollfpr** [comment]
@dukenv0307 @thienlnam It's weird that the information is not accessible from the participant's list but displayed with a direct URL to the profile page. 

https://github.com/Expensify/App/assets/25520267/f41495d0-ec5a-44ec-8dfc-87b102f9dfd0

The profile information is hidden after logging out.

Screenshot 2023-06-14 at 14 37 56

**Comment by mollfpr** [comment]
I think hidden is make more sense, same as after log out.

**Comment by thienlnam** [comment]
>  It's weird that the information is not accessible from the participant's list but displayed with a direct URL to the profile page.

Yeah I agree it might be a weird consistency thing - the only thing is it's because you don't have access to the report which is why. 

We could either:
- Allow you to see it, but strip the details since you no longer should see the participant details
- Just make every page after :/r/reportID just show the `You no longer have access to this chat` screen

**Comment by thienlnam** [comment]
Could you please bring back into open-source? I think we should revisit expected functionality (my bad, should have noted this before we moved to implementation)

File: src/libs/ReportUtils.js [hunk 1]
```diff
@@ -1971,6 +1971,24 @@ function getChatByParticipants(newParticipantList) {
     });
 }
 
+/**
+ * Attempts to find a report in onyx with the provided list of participants. Does not include threads
+ * @param {String} accountID
+ * @param {String} policyID
+ * @returns {Array|undefined}
```
Comment (mollfpr) [inline src/libs/ReportUtils.js:1978]: Minor, I think this should boolean not array.

**Review by mollfpr — COMMENTED** [review]
The removed personal detail only works on admin account. On the second account, the personal detail still available.

https://github.com/Expensify/App/assets/25520267/75fd6b04-0308-4d39-9398-ffb0b59e4f94

File: src/libs/ReportUtils.js [hunk 2]
```diff
@@ -1971,6 +1971,23 @@ function getChatByParticipants(newParticipantList) {
     });
 }
 
+/**
+ * Attempts to find a report in onyx with the provided list of participants. Does not include threads
+ * @param {String} accountID
+ * @returns {boolean|undefined}
+ */
+function getChatByParticipantInclude(accountID) {
+    return !!_.find(allReports, (report) => {
```
Comment (mollfpr) [inline src/libs/ReportUtils.js:1980]: `!!` can be replaced with `_.isEmpty`?

File: src/libs/actions/Policy.js [hunk 3]
```diff
@@ -208,6 +208,7 @@ function removeMembers(accountIDs, policyID) {
     if (accountIDs.length === 0) {
         return;
     }
+
```
Comment (mollfpr) [inline src/libs/actions/Policy.js:211]:
```suggestion
```

Unnecessary changes.

File: src/pages/ProfilePage.js [hunk 4]
```diff
@@ -102,6 +106,7 @@ function ProfilePage(props) {
     }, [accountID]);
 
     const details = lodashGet(props.personalDetails, accountID, {isLoading: ValidationUtils.isValidAccountRoute(accountID)});
+    const isExistingReport = ReportUtils.getChatByParticipantInclude(accountID);
```
Comment (mollfpr) [inline src/pages/ProfilePage.js:109]: Does this mean that you check the personal details, chat with the user or not, and decide to show the information based on that?

Can we remove that personal detail when the user is removed from the workspace instead of checking if the account has chatted with the user?

**Review by mollfpr — CHANGES_REQUESTED** [review]
@dukenv0307 Could you please resolve the conflict?

I have a question regarding checking the personal detail if it's chatted with the account and trying to see if there's another approach. Although, I believe the changes also work, and it's such a minor, so not worrying too much.

File: src/pages/ProfilePage.js (hunk 4 above)
Comment (mollfpr) [reply src/pages/ProfilePage.js:108]: If it's possible.If we do that, we don't need to check with the user with the existing chat, right?

Comment (mollfpr) [reply src/pages/ProfilePage.js:108]: I mean not in the admin account, but instead on the second account that receives pusher notification on removed workspace members.

Comment (mollfpr) [reply src/pages/ProfilePage.js:108]: > Yes, the second account received a pusher notification on the removed workspace members. But how can we do something when received this notification? 

That's what I thought; that's why I'm asking. Maybe you know how to do that 😅

But thanks @dukenv0307 for the update; I'll start testing it.

**Comment by mollfpr** [comment]
<details>
<summary>Screenshot</summary>

Screenshot 2023-06-29 at 18 16 53

</details>

@dukenv0307 The display name is still displayed after reloading the profile page. I think the expected will be showing _Hidden_

Also, could you update the test? 

After step 6 _**Click on any member to open its details page**_, we can add another check to verify the URL is `/a/:accountID` not `/r/:reportID/participants/a/:accountID`.

Your recordings are also outdated from the new expected result.

Thank you!

File: src/libs/ReportUtils.js [hunk 5]
```diff
@@ -2011,6 +2011,25 @@ function getChatByParticipants(newParticipantList) {
     });
 }
 
+/**
+ * Attempts to find a report in onyx with the provided list of participants. Does not include threads
+ * @param {String} accountID
+ * @returns {boolean|undefined}
```
Comment (mollfpr) [inline src/libs/ReportUtils.js:2017]:
```suggestion
 * @returns {boolean}
```

I guess this only return a boolean, right?

**Review by mollfpr — APPROVED** [review]
LGTM and tests well 👍

File: src/libs/ReportUtils.js [hunk 6]
```diff
@@ -2011,6 +2011,25 @@ function getChatByParticipants(newParticipantList) {
...
+/**
+ * Attempts to find a report in onyx with the provided list of participants. Does not include threads
+ * @param {String} accountID
+ * @returns {boolean}
+ */
+function getChatByParticipantInclude(accountID) {
+    return !_.isEmpty(
+        _.find(allReports, (report) => {
+            // If the report has been deleted, or there are no participants (like an empty #admins room) then skip it
+            if (!report || !report.participantAccountIDs || isThread(report) || isArchivedRoom(report)) {
```
Comment (thienlnam) [inline src/libs/ReportUtils.js:2023]: How come we don't look in threads here?

File: src/pages/ProfilePage.js [hunk 7]
```diff
@@ -62,13 +62,17 @@ const propTypes = {
...
+    isLoadingReportData: PropTypes.bool,
+
     ...withLocalizePropTypes,
 };
 
 const defaultProps = {
     // When opening someone else's profile (via deep link) before login, this is empty
     personalDetails: {},
     loginList: {},
+    isLoadingReportData: true,
```
Comment (thienlnam) [inline src/pages/ProfilePage.js:75]: Where does this prop come from?

**Review by thienlnam — CHANGES_REQUESTED** [review]
Left a couple comments on the code, but looking at this now it seems like a workaround. We've depending on checking the onyx data to see if there is a report exists with the participant to determine whether we should show details for them.

This seems like it should be a back-end fix - when someone is removed from a workspace they should sub-sequentially have the personal details removed for other participants in the workspace so that by default the details don't show up.  Thoughts on this @Beamanator @puneetlath

**Comment by Beamanator** [comment]
> Left a couple comments on the code, but looking at this now it seems like a workaround. We've depending on checking the onyx data to see if there is a report exists with the participant to determine whether we should show details for them.
> 
> This seems like it should be a back-end fix - when someone is removed from a workspace they should sub-sequentially have the personal details removed for other participants in the workspace so that by default the details don't show up. Thoughts on this @Beamanator @puneetlath

Ooh that definitely sounds legit - in the BE we can do some kind of re-check to see if the removed user still "knows" the workspace members or not, and try to delete their personal details onyx data if not

I think @puneetlath might have a better idea if we planned to keep the participants route (i think we still wanted to keep it?) though

**Comment by puneetlath** [comment]
I think the changes in this PR that get rid of the old report participants details route and instead route to the new profile page are good. 

However, I agree with @thienlnam that we shouldn't make the changes to the profile page that are done in this PR. The profile page should just show the profile based on whatever details it has in Onyx. And if certain details shouldn't be in Onyx, the back-end should handle that. I like the idea of creating a follow-up back-end issue to handle updating the personalDetails object for the remove-from-workspace scenario.

**Comment by thienlnam** [comment]
Created the internal issue for that here: https://github.com/Expensify/Expensify/issues/297937

**Comment by thienlnam** [comment]
As for this PR @dukenv0307, sorry for all the change ups but could you please update this PR to just update the routes?

File: src/pages/ProfilePage.js (hunk 7 above)
Comment (puneetlath) [reply src/pages/ProfilePage.js:75]: What's the weird error we see in this scenario? Out of curiosity.

File: src/pages/ProfilePage.js [hunk 8]
```diff
@@ -125,7 +128,7 @@ function ProfilePage(props) {
 
     const isCurrentUser = _.keys(props.loginList).includes(login);
     const hasMinimumDetails = !_.isEmpty(details.avatar);
-    const isLoading = lodashGet(details, 'isLoading', false) || _.isEmpty(details);
+    const isLoading = lodashGet(details, 'isLoading', false) || _.isEmpty(details) || props.isLoadingReportData;
```
[the remark this answers is not in the document]
Comment (puneetlath) [reply src/pages/ProfilePage.js:131]: Ahhh I see.

File: src/pages/ProfilePage.js [hunk 9]
```diff
@@ -62,13 +62,17 @@ const propTypes = {
         partnerUserID: PropTypes.string,
     }),
 
+    /** Indicated whether the report data is loading */
```
Comment (puneetlath) [inline src/pages/ProfilePage.js:65]: I think this is clearer, since it seems this key isn't just used to indicate report data is loading. But all the data that openApp gets (including personalDetails). 

```suggestion
    /** Indicates whether the app is loading initial data */
```

**Review by puneetlath — CHANGES_REQUESTED** [review]

**Review by puneetlath — APPROVED** [review]

**Review by thienlnam — APPROVED** [review]
Great, thanks for these changes!

**Comment by thienlnam** [comment]
Was waiting on some tests to pass - all good now!

**Comment by francoisl** [comment]
@thienlnam or @puneetlath any chance one of you could help internal-QA this one please? The QA team was reporting having issues testing [here](https://github.com/Expensify/App/issues/22598#issuecomment-1631681974).

**Comment by puneetlath** [comment]
I think the QA steps are out of date. The removing the user from the workspace shouldn't be a part of it anymore (that's going to be handled from the back-end). I think all this PR changed is routing you to the profile page, which does seem to be working.

**Comment by thienlnam** [comment]
I tested on web and did not run into the error removing a participant so I think we're okay here 
Screenshot 2023-07-12 at 10 22 23 AM

Puneet also tested on web and verified it redirected correctly

File: src/pages/ProfilePage.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (alitoshmatov) [reply src/pages/ProfilePage.js:131]: This change caused profile page to show loading even for invalid account IDs when their invalidity was already clear and no request made to a backend. https://github.com/Expensify/App/issues/32750
